portal/components/react-admin/base/resources/candidateProfile/candidate-profile.js

```diff
@@ -0,0 +1,209 @@
+import React from "react";
+import {
+  List,
+  Datagrid,
+  Pagination,
+  FunctionField,
+  TopToolbar,
+  sanitizeListRestProps,
+  Filter,
+  SearchInput,
+  ExportButton,
+  downloadCSV,
+  TextInput,
+  SelectInput,
+  ReferenceInput,
+
+} from "react-admin";
+import { makeStyles, Typography } from "@material-ui/core";
+import jsonExport from "jsonexport/dist";
+
+const SearchFilter = (props) => {
+  console.log("Props:", props);
+  return (
+    <Filter {...props}>
+      <SearchInput
+        placeholder="Search by Name *"
+        source="name"
+        className="searchBar"
+        alwaysOn
+      />
+    </Filter>
+  );
+};
+
+const exporter = (records) => {
+  const recordsForExport = records.map((record) => {
+    let age = getAge({
+      start: record.DOB,
+      end: null,
+    }).years;
+
+    return {
+      "Candidate Name": record.name ? record.name : "NONE",
+      "Mobile Number": record.mobile_number ? record.mobile_number : "NONE",
+      "Whatsapp Number": record.whatsapp_mobile_number
+        ? record.whatsapp_mobile_number
+        : "NONE",
+      District: record.district_name.name ? record.district_name.name : "NONE",
+      Pincode: record.pincode ? record.pincode : "NONE",
+      "Max Qualification": record.highest_level_qualification
+        .highest_level_qualification_name
+        ? record.highest_level_qualification.highest_level_qualification_name
+        : "NONE",
+      Qualification: record.qualification_detail.qualification_name
+        ? record.qualification_detail.qualification_name
+        : "NONE",
+      Marks: record.final_score_highest_qualification
+        ? record.final_score_highest_qualification
+        : "NONE",
+      "Date of Birth": record.DOB,
+      Age: age,
+      Gender: record.gender.gender_name,
+      "Have you ever been employed": record.ever_employment.employment_status
```

Comment:
@snehar-nd - At no point should we be referencing a nested attribute without optional chaining